homeassistant/components/fjaraskupan/fan.py

```diff
@@ -0,0 +1,197 @@
+            ORDERED_NAMED_FAN_SPEEDS, percentage
+        )
+
+        async with self._device:
+            if preset_mode != self._preset_mode:
+                if preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMERMANUAL)
+                elif preset_mode == PRESET_MODE_AFTER_COOKING_AUTO:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMERAUTO)
+                elif preset_mode == PRESET_MODE_NORMAL:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMEROFF)
+
+            if preset_mode == PRESET_MODE_NORMAL:
+                await self._device.send_fan_speed(int(new_speed))
+            elif preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+                await self._device.send_after_cooking(int(new_speed))
+            elif preset_mode == PRESET_MODE_AFTER_COOKING_AUTO:
+                await self._device.send_after_cooking(0)
+
+        self.coordinator.async_set_updated_data(self._device.state)
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+            await self._device.send_command(COMMAND_AFTERCOOKINGTIMERMANUAL)
+        elif preset_mode == PRESET_MODE_AFTER_COOKING_AUTO:
+            await self._device.send_command(COMMAND_AFTERCOOKINGTIMERAUTO)
+        elif preset_mode == PRESET_MODE_NORMAL:
+            await self._device.send_command(COMMAND_AFTERCOOKINGTIMEROFF)
+
+        self.coordinator.async_set_updated_data(self._device.state)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn the entity off."""
+        await self._device.send_command(COMMAND_STOP_FAN)
+        self.coordinator.async_set_updated_data(self._device.state)
+
+    @property
+    def speed_count(self) -> int:
+        """Return the number of speeds the fan supports."""
+        return len(ORDERED_NAMED_FAN_SPEEDS)
+
+    @property
+    def percentage(self) -> int | None:
+        """Return the current speed."""
+        return self._percentage
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_SET_SPEED | SUPPORT_PRESET_MODE
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if fan is on."""
+        return self._percentage != 0
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return a list of available preset modes."""
+        return [
+            PRESET_MODE_NORMAL,
+            PRESET_MODE_AFTER_COOKING_AUTO,
+            PRESET_MODE_AFTER_COOKING_MANUAL,
+        ]
+
+    def _update_from_device_data(self, data: State | None) -> None:
+        """Handle data update."""
+        if data:
```

Comment:
We can invert this check, set percentage to 0 and return early, then outdent below.